src/Entity/Transaction.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+namespace AppBundle\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Entity(repositoryClass="AppBundle\Repository\TransactionRepository")
+ */
+class Transaction
+{
+    public const PAYBOX_SUCCESS = '00000';
+    public const PAYBOX_CONNECTION_FAILED = '00001';
+    public const PAYBOX_INTERNAL_ERROR = '00003';
+    public const PAYBOX_CARD_NUMBER_INVALID = '00004';
+    public const PAYBOX_CARD_END_DATE_INVALID = '00008';
+    public const PAYBOX_CARD_UNAUTHORIZED = '000021';
+    public const PAYBOX_PAYMENT_PAGE_TIMEOUT = '000030';
+
+    /**
+     * @var int
+     *
+     * @ORM\Column(type="integer", options={"unsigned"=true})
+     * @ORM\Id
+     * @ORM\GeneratedValue(strategy="AUTO")
+     */
+    private $id;
+
+    /**
+     * @var string|null
+     *
+     * @ORM\Column(length=100, nullable=true)
+     */
+    private $payboxResultCode;
+
+    /**
+     * @var string|null
+     *
+     * @ORM\Column(length=100, nullable=true)
+     */
+    private $payboxAuthorizationCode;
+
+    /**
+     * @var array|null
+     *
+     * @ORM\Column(type="json_array", nullable=true)
+     */
+    private $payboxPayload;
+
+    /**
+     * @var \DateTimeImmutable|null
+     *
+     * @ORM\Column(type="datetime_immutable", nullable=true)
+     */
+    private $payboxDateTime;
+
+    /**
+     * @var string|null
+     *
+     * @ORM\Column(unique=true, nullable=true)
+     */
+    private $payboxTransactionId;
+
+    /**
+     * @var string|null
+     *
+     * @ORM\Column(nullable=true)
+     */
+    private $payboxSubscriptionId;
+
+    /**
+     * @var \DateTimeImmutable
+     *
+     * @ORM\Column(type="datetime_immutable")
+     */
+    private $createdAt;
+
+    /**
+     * @var Donation
+     *
+     * @ORM\ManyToOne(targetEntity="AppBundle\Entity\Donation", inversedBy="transactions")
+     */
+    private $donation;
+
+    public function __construct(Donation $donation, array $payboxPayload)
+    {
+        $this->donation = $donation;
+        $this->createdAt = new \DateTimeImmutable();
+        $this->payboxPayload = $payboxPayload;
+        $this->payboxResultCode = $payboxPayload['result'];
+
+        if ($transaction = $payboxPayload['transaction'] ?: false) {
+            $this->payboxTransactionId = $transaction;
+        }
+
+        if (isset($payboxPayload['date'], $payboxPayload['time'])) {
+            [
+                'date' => $date,
+                'time' => $time,
+            ] = $payboxPayload;
+            $this->payboxDateTime = \DateTimeImmutable::createFromFormat(
```

Comment:
what is the value of $date and $time? do you have example?

Reply:
It is the date time when the transaction have been done in paybox. It could be done yesterday but we receive the notification only now.

Reply:
@cacahouwete p48 array N2 
http://www1.paybox.com/wp-content/uploads/2017/08/ManuelIntegrationVerifone_PayboxSystem_V8.0_EN.pdf 

you can do directly: 

```
\DateTimeImmutable::createFromFormat('DDMMYYYYH:i:s', $date.$time) // without substr & co
```

Reply:
When I do it like that I have :
```php
Return value of AppBundle\Entity\Transaction::getPayboxDateTime() must be an instance of DateTimeImmutable or null, boolean returned
```

Caused by the fact createFromFormat return false :s

Reply:
create format need a separation between each element :s

Reply:
But i can do dmYH:i:s